concent_api/core/payments/storage.py

```diff
@@ -0,0 +1,145 @@
+import logging
+from typing import Dict
+from typing import List
+
+from ethereum.transactions import Transaction
+from golem_messages.utils import encode_hex
+#from golem_sci.transactionsstorage import TransactionsStorage
+
+from django.db import transaction
+
+from core.models import GlobalTransactionState
+from core.models import PendingEthereumTransaction
+from core.payments import service as payments_service
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatabaseTransactionsStorage():
+    """
+    Concent custom implementation of TransactionsStorage interface used to store Ethereum transaction data into
+    database using Django models.
+    """
+
+    def __init__(self, *args: List, **kwargs: Dict) -> None:
+        super().__init__(*args, **kwargs)
+
+        if not GlobalTransactionState.objects.filter(pk=0).exists():
+            global_transaction_state = GlobalTransactionState(
+                pk=0,
+                nonce=payments_service.get_transaction_count(),  # pylint: disable=no-value-for-parameter
+            )
+            global_transaction_state.full_clean()
+            global_transaction_state.save()
+
+    @transaction.atomic(using='control')
+    def get_nonce(self) -> int:
+        """
+        Returns the nonce for the next transaction.
+        """
+        global_transaction_state = self._get_locked_global_transaction_state()
+        return global_transaction_state.nonce
+
+    @transaction.atomic(using='control')  # pylint: disable=no-self-use
+    def get_all_tx(self) -> List[Transaction]:
+        """
+        Returns the list of all transactions.
+        """
+        return [
+            Transaction(
+                nonce    = int(ethereum_transaction.nonce),
+                gasprice = int(ethereum_transaction.gasprice),
+                startgas = int(ethereum_transaction.startgas),
+                value    = int(ethereum_transaction.value),
+                v        = int(ethereum_transaction.v),
+                r        = int(ethereum_transaction.r),
+                s        = int(ethereum_transaction.s),
+                to       = encode_hex(ethereum_transaction.to.tobytes()),
+                data     = ethereum_transaction.data.tobytes(),
```

Comment:
Style:)

Reply:
Well, it's technically against our rules but in this case it clearly improves readability. Maybe we should allow for some exceptions?

@dybi What do you think?

Reply:
Yes, exactly, my point here was not to follow old rules but to occasionally break new rules for better readability. But if anyone will insist, I will change it.

@cameel It would be even less readable that in the second part of your example, because there should be no spaces around `=`.

Reply:
I'd prefer for it to be allowed but you have a point. @rwrzesien please raise this issue on the retro.